src/uu/cp/src/cp.rs

```diff
@@ -898,9 +939,32 @@ fn copy_directory(root: &Path, target: &Target, options: &Options) -> CopyResult
     let mut hard_links: Vec<(String, u64)> = vec![];
 
     for path in WalkDir::new(root) {
-        let path = or_continue!(or_continue!(path).path().canonicalize());
+        let p = or_continue!(path);
+        let is_symlink = fs::symlink_metadata(p.path())?.file_type().is_symlink();
+        let path = if options.no_dereference && is_symlink {
+            // we are dealing with a symlink. Don't follow it
+            env::current_dir().unwrap().join(normalize_path(p.path()))
+        } else {
+            or_continue!(p.path().canonicalize())
+        };
+
         let local_to_root_parent = match root_parent {
-            Some(parent) => or_continue!(path.strip_prefix(&parent)).to_path_buf(),
+            Some(parent) => {
+                #[cfg(windows)]
+                {
+                    // On Windows, some pathes are starting with \\?
+                    // but not always, so, make sure that we are consistent for strip_prefix
+                    // See https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file for more info
+                    let parent_can = adjust_canonicalization(parent);
+                    let path_can = adjust_canonicalization(path.clone());
+
+                    or_continue!(path_can.strip_prefix(&parent_can)).to_path_buf()
```

Comment:
There’s a lot of allocation in this branch.  It might be better to make `adjust_canonicalization()` return a `Cow<‘_, Path>`.

Reply:
OK, I will need to have a look on a Windows

Reply:
Actually, it was easy to fix, done :)